src/features/apiDelegation/components/ApiActionBar/ApiActionBar.tsx

```diff
@@ -0,0 +1,222 @@
+import { Button, List, Paragraph, Spinner, Alert, Heading } from '@digdir/design-system-react';
+import { useEffect, useLayoutEffect, useState } from 'react';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { MinusCircleIcon, PlusCircleIcon } from '@navikt/aksel-icons';
+
+import { ActionBar, type ActionBarProps } from '@/components';
+import { ErrorCode, getErrorCodeTextKey } from '@/resources/utils/errorCodeUtils';
+import { type DelegableApi } from '@/rtk/features/apiDelegation/delegableApi/delegableApiSlice';
+import { useDelegationCheckMutation } from '@/rtk/features/apiDelegation/apiDelegationApi';
+import type { ResourceReference } from '@/rtk/features/apiDelegation/apiDelegationApi';
+import { getCookie } from '@/resources/Cookie/CookieMethods';
+import type { DelegationAccessResult } from '@/dataObjects/dtos/resourceDelegation';
+import ScopeList from '@/components/ScopeList/ScopeList';
+
+import classes from './ApiActionBar.module.css';
+
+export interface ApiActionBarProps extends Pick<ActionBarProps, 'color'> {
+  /** Defines the functionality and behaviour of the actionbar, whether it is used to add a new api or to remove one that was previously added */
+  variant: 'add' | 'remove';
+  /** Used when the variant equals 'add'. The callback that is triggered when the user clicks the add button, if they pass the delegationCheck. */
+  onAdd?: () => void;
+  /** Used when the variant equals 'remove'. The callback that is triggered when the user clicks the remove button. */
+  onRemove?: () => void;
+  /** The API to be represented */
+  api: DelegableApi;
+  /** Whether or not to run a delegationCheck on initialization. Used for deep links with pre-chosen API's */
+  initWithDelegationCheck?: boolean;
+}
+
+export const ApiActionBar = ({
+  variant,
+  onAdd,
+  onRemove,
+  api,
+  initWithDelegationCheck = false,
+}: ApiActionBarProps) => {
+  const [open, setOpen] = useState(false);
+  const [actionBarColor, setActionBarColor] = useState<'success' | 'danger' | 'neutral'>(
+    variant === 'remove' ? 'success' : 'neutral',
+  );
+  const { t } = useTranslation('common');
+  const resourceRef: ResourceReference = { resource: api.authorizationReference };
+  const partyId = getCookie('AltinnPartyId');
+
+  const [checkCanDelegate, { data: accessResult, error, isLoading: isUpdating }] =
```

Comment:
imo isLoading is a clearer phrasing than isUpdating so I don't know why you converted it do isUpdating

Reply:
It was mostly because queries have isLoading and isFetching which represent the initial loading and the refetching of data respectively, while mutations only have isLoading representing both. rtkQuery's documentation suggested renaming it to isUpdating locally to make that more clear and I figured I'd do the same, but I guess in our case that distinction doesn't really matter so isLoading is better.
I'll change it 👍